### PR TITLE
4-create-function-that-generates-a-list-of-dictionaries-containing-model-metadata-from-deployed-endpoints

### PR DESCRIPTION
**Pull request by RussellJimmies**
#4

### CODE REVIEW

File: models_metadata.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        model_endpoint_metadata['job_name'] = job.display_name
+
+        model_endpoints_metadata.append(model_endpoint_metadata)
+
+    return model_endpoints_metadata 
+        
+if __name__ == "__main__":
+    model_endpoints_metadata = generate_model_endpoints_metadata(ml_client)
+    with open("model_endpoints_metadata.json", "w") as outfile:
+        json.dump(model_endpoints_metadata, outfile)
```
Comment (rngadam) [inline models_metadata.py:79]: json.dump(model_endpoints_metadata, outfile, indent=4)

store these json in a nicely formatted json that is human readable.

File: .env.template [hunk 2]
```diff
@@ -0,0 +1,3 @@
+NACHET_SUBSCRIPTION_ID=
+NACHET_RESOURCE_GROUP=
+NACHET_WORKSPACE=
```
Comment (rngadam) [inline .env.template:3]: https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

File: models_metadata.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+        # Retrieve deployment's model (from filePath)
+        model_filepath = deployment.model
+        pattern = re.compile(r"models/([^/]+)/versions/(\d+)")
+        match = pattern.search(model_filepath)
+        if match:
+            model_name = match.group(1)  
+            model_version = match.group(2)  
+        else:
+            raise Exception("No match found.")
```
Comment (rngadam) [inline models_metadata.py:58]: create a custom exception. each layer/module should have its own exception.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: models_metadata.py (hunk 1 above)
Comment (RussellJimmies) [reply models_metadata.py:79]: Done

File: .env.template (hunk 2 above)
Comment (RussellJimmies) [reply .env.template:3]: Done

File: models_metadata.py (hunk 3 above)
Comment (RussellJimmies) [reply models_metadata.py:58]: Done

**Review by rngadam — APPROVED** [review]

**Comment by rngadam** [comment]
added @amirardalan9473 as reviewer

**Pull request merged**